cli/dstack/core/repo/remote.py

```diff
@@ -90,31 +68,14 @@ def path(self, sep: str = ".") -> str:
             ]
         )
 
-
-class Repo(ABC):
-    def __init__(self, repo_ref: RepoRef, repo_data: RepoData):
-        self.repo_ref = repo_ref
-        self.repo_data = repo_data
-
-    @property
-    def repo_id(self) -> str:
-        return self.repo_ref.repo_id
-
-    @property
-    def repo_user_id(self) -> str:
-        return self.repo_ref.repo_user_id
-
-    @property
-    def repo_spec(self) -> "RepoSpec":
```

Comment:
Why dropping `repo_spec` in favor of `RepoSpec.from_repo()`? Seems like pointless refactoring...

Reply:
It resolves the circular dependency: `Repo.repo_spec` and `RepoSpec.repo`